### PR TITLE
[bitnami/grafana-operator] Update the grafana-operator chart to version 4.x release.

### LINKED ISSUE #7704: [bitnami/grafana-operator] Add support for Kubernetes 1.22+
**Which chart**:
bitnami/grafana-operator

**Is your feature request related to a problem? Please describe.**
Currently it is not possible to install grafana-operator on Kubernetes 1.22+ because it does not yet use `apiextensions.k8s.io/v1` for its CRDs and `apiextensions.k8s.io/v1beta1` is not available with 1.22 anymore.

**Describe the solution you'd like**
Today a new version (3.10.4: https://github.com/grafana-operator/grafana-operator/releases/tag/v3.10.4) of the grafana-operator was release that provides support for Kubernetes 1.22+. An upgrade to that version could fix this chart.

### LINKED ISSUE #8333: Change K8s API from apiextensions.k8s.io/v1beta1 to apiextensions.k8s.io/v1
**Which chart**:
grafana-operator

**Describe the bug**
After trying to install Helm chart I get the following error:
```
Error: INSTALLATION FAILED: failed to install CRD crds/grafanadashboards.yaml: unable to recognize "": no matches for kind "CustomResourceDefinition" in version "apiextensions.k8s.io/v1beta1"
```

**To Reproduce**
```
helm repo add bitnami https://charts.bitnami.com/bitnami
helm install grafana-operator bitnami/grafana-operator
```

**Expected behavior**
Helm charts installs without errors.

**Version of Helm and Kubernetes**:

- Output of `helm version`:

```
version.BuildInfo{Version:"v3.7.1", GitCommit:"1d11fcb5d3f3bf00dbe6fe31b8412839a96b3dc4", GitTreeState:"clean", GoVersion:"go1.16.9"}
```

- Output of `kubectl version`:

```
Client Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.4", GitCommit:"b695d79d4f967c403a96986f1750a35eb75e75f1", GitTreeState:"clean", BuildDate:"2021-11-17T15:48:33Z", GoVersion:"go1.16.10", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.3", GitCommit:"c92036820499fedefec0f847e2054d824aea6cd1", GitTreeState:"clean", BuildDate:"2021-10-27T18:35:25Z", GoVersion:"go1.16.9", Compiler:"gc", Platform:"linux/amd64"}
```

- Output of `minikube version`:

```
minikube version: v1.24.0
```

**Additional context**
As it's said in [Kubernetes docs reference](https://kubernetes.io/docs/reference/using-api/deprecation-guide/#customresourcedefinition-v122):

> apiextensions.k8s.io/v1beta1 API version of CustomResourceDefinition is no longer served as of v1.22

Solution would be to change existing API **apiextensions.k8s.io/v1beta1** to **apiextensions.k8s.io/v1**. It seems the same applies to the **kong** chart.

### PR DESCRIPTION
**Pull request by Cellebyte**
**Description of the change**

It updates the CRD's from the project and the operator to the v4.0.x release.

**Benefits**

Better validation of the CRD Resources via API-Server and third-party tools.
Reduced rbac roles.
Leader Election for better HA support of the operator.

**Possible drawbacks**

**Applicable issues**

  - Fixes #7704
  - Waits for the following Pull Requests
    - https://github.com/grafana-operator/grafana-operator/pull/559
    - https://github.com/grafana-operator/grafana-operator/pull/601
    - https://github.com/bitnami/bitnami-docker-grafana/issues/17
  - Incorporates the changes of #7896 
  - Incorporates the changes of #8513
  - Fixes #8333

**Additional information**

**Checklist** 

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. [bitnami/<name_of_the_chart>])

### CODE REVIEW

**Comment by carrodher** [comment]
Hi @Cellebyte, is this PR ready for review, or are you planning more tests/changes on your side? I'm asking because of the `DRAFT:` that appears in the title. Feel free to ping me once it is ready for review. Thanks!

**Comment by Cellebyte** [comment]
@carrodher we still need to wait until the upstream changes are implemented.
Then build a new operator version and then bump it here and merge it.

**Comment by carrodher** [comment]
Thanks for the changes @Cellebyte, I just pinged @juan131 in a related PR he already created (see above link) so you can work together if needed

**Comment by Cellebyte** [comment]
@carrodher I already added a comment. What from my perspective makes more sense.

**Comment by Cellebyte** [comment]
@carrodher @juan131 the Pull Request is now able to get reviewed.

It incorporates all the changes @juan131 did.

**Comment by javsalgar** [comment]
Hi,

Could you check the merge issue?

**Comment by Cellebyte** [comment]
@javsalgar resolved the merge conflict.
We need to get the grafana image fixed before we can merge it here.

**Comment by Cellebyte** [comment]
@javsalgar still waiting on the small changes in the grafana image so that we can progress here.

**Comment by NissesSenap** [comment]
@Cellebyte if it's 4.1.0 you are waiting for it's fixed now.
https://quay.io/repository/grafana-operator/grafana-operator?tab=tags
https://github.com/grafana-operator/grafana-operator/releases/tag/v4.1.0

**Comment by Cellebyte** [comment]
@NissesSenap I am waiting for 
* [https://github.com/bitnami/bitnami-docker-grafana/issues/17](https://github.com/bitnami/bitnami-docker-grafana/issues/17)

**Comment by Cellebyte** [comment]
With my new Pull Request here
* https://github.com/bitnami/bitnami-docker-grafana/pull/18

The issues are resolved I think you need internal validation to make sure that I am not introducing breaking changes.
After this Pull Request gets merged we can proceed here.

**Review by joancafom — CHANGES_REQUESTED** [review]
Thanks for your changes. I left some comments while the issue you reported is reviewed and addressed, could you please take a look at them?

File: bitnami/grafana-operator/templates/rbac.yaml
Comment (juan131) [inline bitnami/grafana-operator/templates/rbac.yaml:46]: Could you use 2 space indentation for arrays? It's just a matter of conventions with the rest of the files in the repo.

Comment (Cellebyte) [reply bitnami/grafana-operator/templates/rbac.yaml:46]: Yep

File: bitnami/grafana-operator/README.md
Comment (Cellebyte) [reply bitnami/grafana-operator/README.md:21]: Sadly not because the `crds` directory is not templated.

Comment (tomberget) [reply bitnami/grafana-operator/README.md:21]: Not saying that you should, but it would be possible to move them to the `templates` directory, and then template them there. That would satisfy both the "old" and the "new" :)

Comment (Cellebyte) [reply bitnami/grafana-operator/README.md:21]: @tomberget I had a lot of issues with helm charts templating crds. It is still not mature. Also if the chart relies on the crds to be created beforehand.

Comment (Cellebyte) [reply bitnami/grafana-operator/README.md:21]: It also bricked a lot of our `helm upgrade` commands in our 100+ clusters when a chart decided to use templated crds.

**Comment by juan131** [comment]
@joancafom are we ready to go with this?

**Comment by Cellebyte** [comment]
@juan131 please do not merge until we have fixed the grafana image :D

File: bitnami/grafana-operator/values.yaml
Comment (miguelaeh) [inline bitnami/grafana-operator/values.yaml:316]: Could you please elaborate on why you deleted the `fsGroup` ? It is used to configure the permissions in the volumes in some K8s clusters.

Comment (Cellebyte) [reply bitnami/grafana-operator/values.yaml:316]: because the kubernetes documentation says that for the individual containers `securityContext` the `fsGroup` attribute is not supported. -> 
* [podSecurityContext](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.22/#podsecuritycontext-v1-core)
* [containerSecurityContext](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.22/#securitycontext-v1-core)

That is also why it got removed from the `grafanas` CRD in the grafana-operator.

Comment (miguelaeh) [reply bitnami/grafana-operator/values.yaml:316]: yes, that's righ, I just found the issues please check the thread in the [original issue](https://github.com/bitnami/bitnami-docker-grafana/issues/17)

Comment (Cellebyte) [reply bitnami/grafana-operator/values.yaml:316]: https://github.com/grafana-operator/grafana-operator/issues/639

**Review by yardenshoham — APPROVED** [review]

**Comment by javsalgar** [comment]
@Cellebyte could you try merging the changes I added in this PR so we don't bump the chart twice? https://github.com/bitnami/charts/pull/8513

**Comment by Cellebyte** [comment]
@javsalgar will do that :)

**Comment by Cellebyte** [comment]
@javsalgar could you regenerate the readme from the new `values.yaml` file.

**Comment by javsalgar** [comment]
@Cellebyte updated

**Comment by javsalgar** [comment]
@Cellebyte I want to launch our automated test suite and, in case it succeeds, proceed with the automatic merge, any blockers?

**Comment by Cellebyte** [comment]
@javsalgar From my side it should be done.

**Comment by Cellebyte** [comment]
Finally it’s done thanks to all of you :)

**Pull request merged**
